src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingRegistry.java

```diff
@@ -135,12 +139,30 @@ public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate
                 updateThing(thing);
             }
         } catch (CloudServiceInteractionException e) {
-            List<String> certIds = thing.getAttachedCertificateIds();
-            if (certIds.contains(certificate.getCertificateId())) {
+            Map<String, Instant> certIds = thing.getAttachedCertificateIds();
+            if (certIds.containsKey(certificate.getCertificateId())) {
                 return true;
             }
             throw e;
         }
         return false;
     }
+
+    private Thing dtoToThing(ThingV1 dto) {
```

Comment:
Now I get it... If ThingV1 is a dto could we make  the name `ThingV1DTO` as a name?

Reply:
+1, we put it in a dto package, but you don't see that when reading the code. Good suggestion

Reply:
Renamed DTOs